CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/ImportPane.java

```diff
@@ -115,34 +106,16 @@ public ImportPane(final ImportTopComponent importTopComponent, final ImportContr
             showSchemaAttributesCheckBox.setSelected(newPreference);
         });
 
-        // show schema attributes menu item
-        showSchemaAttributesItem = new MenuItem("Show all schema attributes", showSchemaAttributesCheckBox);
-        showSchemaAttributesItem.setOnAction((ActionEvent event) -> {
-            // ignore if the checkbox was clicked
-            if (!userClickedTheCheckboxFirst[0]) {
-                final boolean newPreference = !importExportPrefs.getBoolean(
-                        ImportExportPreferenceKeys.SHOW_SCHEMA_ATTRIBUTES,
-                        ImportExportPreferenceKeys.DEFAULT_SHOW_SCHEMA_ATTRIBUTES);
-                importExportPrefs.putBoolean(ImportExportPreferenceKeys.SHOW_SCHEMA_ATTRIBUTES, newPreference);
-                importController.setShowAllSchemaAttributes(newPreference);
-                importController.setClearManuallyAdded(false);
-                importController.setDestination(sourcePane.getDestination());
-                final int saveResultsItemIndex = optionsMenu.getItems().indexOf(showSchemaAttributesItem);
-                ((CheckBox) optionsMenu.getItems().get(saveResultsItemIndex).getGraphic()).setSelected(newPreference);
-            }
-            userClickedTheCheckboxFirst[0] = false;
-        });
```

Comment:
It looks like this code is what's supposed to happen when the 'Show all schema attributes' item is (un)ticked so it probably needs to be retained in some way

Reply:
I think its for the item variant of showSchemaAttributes. so showSchemaAttributesItem not showSchemaAttributesCheckBox. The item variant was only needed if it was in the options menu. I was thinking it was for if the user clicked on the show schema attribute label in the menu instead of the check box. I think it was just a safety net so the check box would get ticked even if they didn't specifically tick it. I can put all the code I deleted back in if you think it could be doing something else however?

Reply:
@Atlas139mkm your right, it was if they clicked the label and I added that bit of code so I remember it. OK for it to be deleted now that its a button.